concepts/Common/Integration Guides/40 Bind DevExtreme components to GraphQL/40 Bind DevExtreme components to GraphQL.md

```diff
@@ -0,0 +1,335 @@
+            query: Query, 
+            mutation: Mutation, 
+        }); 
+
+## Step 4: Connect the frontend to the backend
+
+The [@apollo/client](https://www.apollographql.com/docs/react/) library enables us to use GraphQL on the client.
+
+Import the following objects from the library into the `App.tsx` file of your Vite project:
+
+    <!--tab: App.tsx-->
+    import { gql, useApolloClient, useMutation } from '@apollo/client';
+
+The `gql` object allows you to form GraphQL queries. The `useMutation` function allows you to map these queries to GraphQL mutations.
+
+If you have difficulties with this section, read the [Apollo documentation](https://www.apollographql.com/docs/react/api/link/introduction/) for a full overview of the library's data flow. Consult the [source code](https://github.com/DevExpress-Examples/devextreme-datagrid-graphql-integration/blob/master/client/vite-react/src/App.tsx) of the tutorial repository to view the solution in full.
+
+1. Create a GraphQL query template for each data type:
+
+        <!--tab: App.tsx-->
+        const CustomersQuery = gql`
+        query{
+        Customers {
+            Value,
+            Text
+        }
+        }`;
+
+2. Create a GraphQL query template for each mutation type:
+
+        <!--tab: App.tsx-->
+        const InsertOrderMutation = gql`
+        mutation($CustomerID: String, $OrderDate: String, $Freight: Float, $ShipCountry: String, $ShipVia: Int){
+        InsertOrder(CustomerID: $CustomerID, OrderDate: $OrderDate, Freight: $Freight, ShipCountry: $ShipCountry, ShipVia: $ShipVia){
+            OrderID,
+            CustomerID,
+            OrderDate,
+            Freight,
+            ShipCountry,
+            ShipVia
+        }
+        }`
+
+3. Within the main function, enable Apollo and import these mutation queries:
+
+        <!--tab: App.tsx-->
+        export default function App() {
+            const appoloClient = useApolloClient();
+            const [InsertOrder] = useMutation(InsertOrderMutation);
+            <...>
+        }
+   
+5. Modify the application's `ordersData` CustomStore to work with GraphQL. Define custom CRUD handlers for our mutations:
+
+        <!--tab: App.tsx-->
+        load: () => sendRequest(`Orders`).then(data => _.cloneDeep(data)),  // The .cloneDeep function creates a mutable copy of our data
+        insert: (values) => sendRequest(`InsertOrder`, { 
+          values, 
+        }), 
+        update: (key, values) => sendRequest(`UpdateOrder`, { 
+        key, 
+          values, 
+        }), 
+        remove: (key) => sendRequest(`DeleteOrder`, { 
+          key, 
+        }), 
+
+6. Enclose these queries in a conditional `switch` statement to improve readability:
+
+        <!--tab: App.tsx-->
+        const sendRequest = (query: string, data: KeyValues = {}) => { 
+            switch (query) { 
+              case 'Orders': 
+                return appoloClient.query({query: OrdersQuery}).then(response => response.data.Orders);
```

Comment:
After reading this up and down a few times, I'm not convinced that we have the best approach here. Apollo Client normally has various "standard" methods of interacting with data, for instance `useQuery` with its support for `loading` and `error` states. Not even mentioning subscriptions...

I think it would be better to show a sample that fully accepts these patterns and shows Apollo users how to make them work with our components. Otherwise it will appear as if our data layer isn't really compatible with Apollo Client -- unless perhaps that's the truth?